modules/auxiliary/server/capture/ldap.rb

```diff
@@ -0,0 +1,179 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::LDAP::Server
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Authentication Capture: LDAP',
+        'Description' => %q{
+          This module mocks an LDAP service to capture authentication
+          information of a client trying to authenticate against an LDAP service
+        },
+        'Author' => 'JustAnda7',
+        'License' => MSF_LICENSE,
+        'Action' => [
+          [ 'Capture', { 'Description' => 'Run an LDAP capture server' } ]
+        ],
+        'PassiveActions' => [ 'Capture' ],
+        'DefaultActions' => 'Capture',
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptAddress.new('SRVHOST', [ true, 'The localhost to listen on.', '0.0.0.0' ]),
+        OptPort.new('SRVPORT', [ true, 'The local port to listen on.', '389' ]),
+        OptString.new('Authentication', [ false, 'The type of authentication used by the client.', 'Simple' ])
```

Comment:
Should this be an option?  If more that one auth format is supported would you need to set this?

Standard options are always registered in all `CAPS`. Also an `OptEnum` would be appropriate here to enforce specific values.

```suggestion
        OptEnum.new('AUTHENTICATION', [ false, 'The type of authentication used by the client.', %w[ Simple ])
```

Reply:
I will remove this option for now as discussed. We can add it if required in future.